src/DataSourceProvider.vue

```diff
@@ -185,22 +186,28 @@ export default {
 
       if (this.selectedDataSource.accessRules === null || !this.selectedDataSource.accessRules.length) {
         this.securityEnabled = false;
+        this.missingAccessTypes = this.widgetData.accessRules[0] && this.widgetData.accessRules[0].type;
```

Comment:
See https://github.com/Fliplet/fliplet-widget-form-builder/pull/350#discussion_r528651702 before I review this as this PR seems to have a similar logic of only checking the first access rule.